### ISSUE TITLE
Arrays have no primitive page in rustdoc

### ISSUE THREAD

**Issue by SimonSapin** [rust-lang/rust#22714]
E.g. like http://doc.rust-lang.org/std/core/primitive.slice.html but for `[T; N]`.

cc #15318, #15654, @steveklabnik

**Comment by steveklabnik** [comment]
```
diff --git a/src/libcore/array.rs b/src/libcore/array.rs
index e8f6e31..fc80dcb 100644
--- a/src/libcore/array.rs
+++ b/src/libcore/array.rs
@@ -14,6 +14,8 @@
 
 #![unstable(feature = "core")] // not yet reviewed
 
+#![doc(primitive = "array")]
+
 use clone::Clone;
 use cmp::{PartialEq, Eq, PartialOrd, Ord, Ordering};
 use fmt;
```

did not actually add the correct page, so I'm not sure. @alexcrichton , I think you helped me with something similar, what should I do to make rustdoc happy here?

**Comment by tomjakubowski** [comment]
My experience with @steveklabnik patch (which included some rustdoc changes, [here][klabpatch]) was that it created a page for the `array` primitive at `core::array`, but the page did not have any impls. It looks like `FixedVector` (`[T; n]`) impls are pushed into the primitive page for `PrimitiveType::Slice`. 

[klabpath][https://github.com/steveklabnik/rust/commit/9d4611119a744881f1494d5c121201dbc7106109]

**Comment by tomjakubowski** [comment]
This was closed by #23633
